azurerm/internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -79,46 +79,98 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
-						"no_public_ip": {
-							Type:     pluginsdk.TypeBool,
-							ForceNew: true,
+						"aml_workspace_id": {
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"customer_managed_key": {
+							Type:     pluginsdk.TypeList,
 							Optional: true,
-							AtLeastOneOf: []string{"custom_parameters.0.no_public_ip", "custom_parameters.0.public_subnet_name",
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
+							Computed: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"key_source": {
```

Comment:
key is in the block name already
```suggestion
									"source": {
```